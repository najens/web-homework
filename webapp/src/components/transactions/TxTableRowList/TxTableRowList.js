import React from 'react'
import { arrayOf, string, bool, number, shape } from 'prop-types'
import { getMonthAndDayFromTimestamp } from '../../../helpers/dateTime'
import { convertCentsToUsDollars } from '../../../helpers/currency'
import TxTableRowListItem from '../TxTableRowListItem'

export default function TxTableRowList ({ data }) {
  return data.map((tx, index) => {
    const {
      id,
      insertedAt,
      category,
      user: { firstName, lastName },
      description,
      merchant: { name: merchantName },
      amount
    } = tx

    const categoryName = category ? category.name : 'None'
    const userName = `${firstName} ${lastName}`
    const date = getMonthAndDayFromTimestamp(insertedAt)
    const dollarAmount = convertCentsToUsDollars(amount)
    const indexIsOdd = index % 2 !== 0
    return (
      <TxTableRowListItem
        categoryName={categoryName}
        date={date}
        description={description}
        dollarAmount={dollarAmount}
        id={id}
        indexIsOdd={indexIsOdd}
        key={`transaction-${id}`}
        merchantName={merchantName}
        userName={userName}
      />
    )
  })
}

TxTableRowList.propTypes = {
  data: arrayOf(shape({
    amount: number,
    category: shape({
      name: string
    }),
    credit: bool,
    debit: bool,
    description: string,
    id: string,
    inserted_at: string,
    merchant: shape({
      name: string
    }),
    user: shape({
      firstName: string,
      lastName: string
    })
  }))
}
