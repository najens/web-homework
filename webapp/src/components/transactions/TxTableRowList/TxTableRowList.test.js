import React from 'react'
import { create } from 'react-test-renderer'
import TxTableRowList from './TxTableRowList'

// Only testing render since this component has no business logic
describe('TxTableRowList component', () => {
  it('renders correctly', () => {
    const data = [
      {
        __typename: 'Transaction',
        id: '782a1d8d-08dc-4ef8-bab0-d922038971a2',
        insertedAt: '2021-03-12T20:00:29',
        user: {
          __typename: 'User',
          firstName: 'Bobby',
          lastName: 'Smith'
        },
        category:{
          __typename: 'Category',
          name: 'Medical'
        },
        description: 'Hot Dogs',
        merchant: {
          __typename: 'Merchant',
          name: 'Taco Bell'
        },
        debit: true,
        credit: false,
        amount: 304
      }
    ]

    const testRenderer = create(<TxTableRowList data={data} />)
    expect(testRenderer.toJSON()).toMatchSnapshot()
  })
})
