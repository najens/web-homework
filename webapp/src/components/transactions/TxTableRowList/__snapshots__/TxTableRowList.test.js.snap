// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxTableRowList component renders correctly 1`] = `
.emotion-6 {
  font-size: 12px;
  line-height: 2;
}

.emotion-0 {
  border: 1px solid #b4c1ec;
  padding: 0px 6px;
}

<tr
  className="emotion-6"
  data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2"
>
  <td
    className="emotion-0"
    data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-date"
  >
    Mar 12
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-userName"
  >
    Bobby Smith
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-description"
  >
    Hot Dogs
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-merchantName"
  >
    Taco Bell
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-category"
  >
    Medical
  </td>
  <td
    className="emotion-0"
    data-testid="transaction-782a1d8d-08dc-4ef8-bab0-d922038971a2-amount"
  >
    $3.04
  </td>
</tr>
`;
